test/SolidityEndToEndTest.cpp

```diff
@@ -3181,6 +3181,27 @@ BOOST_AUTO_TEST_CASE(pass_dynamic_arguments_to_the_base_base_with_gap)
 	BOOST_CHECK(callContractFunction("m_i()") == encodeArgs(4));
 }
 
+BOOST_AUTO_TEST_CASE(constant_functions)
+{
+	char const* sourceCode = R"(
+		contract Foo {
+			function getX() constant returns (uint r) { r = x; }
+			uint x = 56;
+	})";
+	compileAndRun(sourceCode);
+	BOOST_CHECK(callContractFunction("getX()") == encodeArgs(56));
+}
+
+BOOST_AUTO_TEST_CASE(const_state_variable)
+{
+	char const* sourceCode = R"(
+		contract Foo {
+			function getX() constant returns (uint r) { return x; }
+			constant uint x = 56;
```

Comment:
ah, I took it from the description of the story at pivotal tracker. I'll change it